### PR TITLE
Add integration test against Janus

### LINKED ISSUE #13: Integration test against Janus
After building our JNI library for the host platform (#3), we should leverage [testcontainers](https://java.testcontainers.org/) to write new unit tests that spin up a leader, helper, and collector, and confirm that reports from the client get aggregated properly.

### PR DESCRIPTION
**Pull request by divergentdave**
This adds an integration test against Janus, making use of the Java testcontainers library to manage the lifetime of our Janus interoperation test containers. This closes #13.

### CODE REVIEW

File: divviup/src/test/java/org/divviup/android/JanusIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+            propagateLogs(collector, "collector");
+        }
+
+        assertNotNull(result);
+        assertEquals(expectedResult, result);
+    }
+
+    private static TaskId randomTaskId() {
+        byte[] bytes = new byte[32];
+        new Random().nextBytes(bytes);
```
Comment (inahga) [inline divviup/src/test/java/org/divviup/android/JanusIntegrationTest.java:269]: Possibly nit, can leave it if you don't think it's pertinent: `java.util.Random` [is not a CSPRNG](https://docs.oracle.com/javase/8/docs/api/java/util/Random.html).

For practical usage in these tests it probably doesn't matter, but I would be worried about someone reading these tests for reference on how to use the library, and copying over usage of `java.util.Random`.

File: divviup/src/test/java/org/divviup/android/JanusIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,439 @@
...
+import java.util.Random;
+
+public class JanusIntegrationTest {
+    /** @noinspection SpellCheckingInspection */
+    private static final DockerImageName JANUS_INTEROP_AGGREGATOR = DockerImageName.parse(
+            "us-west2-docker.pkg.dev/divviup-artifacts-public/janus/janus_interop_aggregator@sha256:8cc873f7a8be459fe2dbecdf78561806b514ac98b4d644dc9a7f6bb25bb9df02"
+    );
+    /** @noinspection SpellCheckingInspection */
+    private static final DockerImageName JANUS_INTEROP_COLLECTOR = DockerImageName.parse(
+            "us-west2-docker.pkg.dev/divviup-artifacts-public/janus/janus_interop_collector@sha256:982110bc29842639355830339b95fac77432cbbcc28df0cd07daf91551570602"
```
Comment (inahga) [inline divviup/src/test/java/org/divviup/android/JanusIntegrationTest.java:49]: Q: why are these SHAs rather than tagged versions?

**Comment by divergentdave** [comment]
The Java version of testcontainers has some interesting features that might be worth copying in Janus. First off, they start up a daemon called [Ryuk](https://github.com/testcontainers/moby-ryuk) in a container that does last-resort container cleanup if the test process disconnects and doesn't clean up after itself for whatever reason. Second, they have a feature that allows exposing a TCP server on a host port within a container network. Under the hood, this works by running sshd in a container, attached to the correct Docker network, and running ssh on the host with reverse tunnel options set. This could allow for new combinations of in-process an containerized aggregators in our integration tests.

File: divviup/src/test/java/org/divviup/android/JanusIntegrationTest.java (hunk 2 above)
Comment (divergentdave) [reply divviup/src/test/java/org/divviup/android/JanusIntegrationTest.java:49]: I did it for consistency with divviup-ts, but I now see that we have both the tag and SHA over there. I'll update this to match.

**Review by tgeoghegan — APPROVED** [review]

**Pull request merged**